API.md

```diff
@@ -0,0 +1,337 @@
+| **Capacity** | int | Gets or sets the capacity |
+| **Length** | long | Gets the number of bytes written to this stream. |
+| **Position** | long | Gets the current position in the stream |
+| **CanRead** | bool | Whether the stream can currently read |
+| **CanSeek** | bool | Whether the stream can currently seek |
+| **CanTimeout** | bool | Always false |
+| **CanWrite** | bool | Whether the stream can currently write |
+| **Disposed** | bool |  |
+| **ReadTimeout** | int |  |
+| **WriteTimeout** | int |  |
+## Constructors
+
+| Name | Summary |
+|---|---|
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager)** | Allocate a new RecyclableMemoryStream object. |
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager, Guid id)** | Allocate a new RecyclableMemoryStream object. |
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager, string tag)** | Allocate a new RecyclableMemoryStream object |
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager, Guid id, string tag)** | Allocate a new RecyclableMemoryStream object |
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager, string tag, int requestedSize)** | Allocate a new RecyclableMemoryStream object |
+| **RecyclableMemoryStream([RecyclableMemoryStreamManager](#recyclablememorystreammanager-class) memoryManager, Guid id, string tag, int requestedSize)** | Allocate a new RecyclableMemoryStream object |
+## Methods
+
+| Name | Returns | Summary |
+|---|---|---|
+| **Close()** | void | Equivalent to Dispose |
+| **GetBuffer()** | byte[] | Returns a single buffer containing the contents of the stream.<br>The buffer may be longer than the stream length. |
+| **Read(byte[] buffer, int offset, int count)** | int | Reads from the current position into the provided buffer |
+| **ReadByte()** | int | Reads a single byte from the current position in the stream. |
+| **SafeRead(byte[] buffer, int offset, int count, Int32& streamPosition)** | int | Reads from the specified position into the provided buffer |
+| **SafeReadByte(Int32& streamPosition)** | int | Reads a single byte from the specified position in the stream. |
+| **Seek(long offset, SeekOrigin loc)** | long | Sets the position to the offset from the seek location |
+| **SetLength(long value)** | void | Sets the length of the stream |
+| **ToArray()** | byte[] | Returns a new array with a copy of the buffer's contents. You should almost certainly be using GetBuffer combined with the Length to <br>access the bytes in this stream. Calling ToArray will destroy the benefits of pooled buffers, but it is included<br>for the sake of completeness. |
```

Comment:
I have seen benefits from using this library to build up a long “stream” from lots of small chunks, but the library I am required to pass the output to only takes a byte[].  Is ToArray my best option?  Could I do some sort of span thing? Any chance this library could integrate with renting arrays?

Reply:
If the API only takes a byte[] with no length parameter, you might be stuck using ToArray, which stinks, because you lose all the pooling benefits. I'm not sure there is another option--it all comes down to the APIs you have to call, even with Span.